amd_openvx_extensions/amd_custom/source/custom_layer.cpp

```diff
@@ -0,0 +1,244 @@
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_DATA_TYPE, &out_data_type, sizeof(out_data_type)));
+    data->backend = customBackend::CPU;   //default based on compiler flag
+
+    if (parameters[2])
+      ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[2], &data->backend, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+#if ENABLE_HIP
+    if (data->backend == customBackend::GPU) { 
+        ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_STRIDE_GPU, input_strides, sizeof(input_strides)));
+        ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_STRIDE_GPU, output_strides, sizeof(output_strides)));
+        ERROR_CHECK_STATUS(vxQueryNode(node, VX_NODE_ATTRIBUTE_AMD_HIP_STREAM, &data->hipstream, sizeof(data->hipstream)));
+    }
+    else {
+        data->backend = customBackend::CPU;   //default for CPU
+        input_strides[0] = sizeof(in_data_type);
+        input_strides[1] = input_strides[0]* input_dims[0];
+        input_strides[2] = input_strides[1]* input_dims[1];
+        input_strides[3] = input_strides[2]* input_dims[2];
+        output_strides[0] = sizeof(out_data_type);
+        output_strides[1] = output_strides[0]* output_dims[0];
+        output_strides[2] = output_strides[1]* output_dims[1];
+        output_strides[3] = output_strides[2]* output_dims[2];
+    }
+#else
+    data->backend = customBackend::CPU;   //default for CPU
+    input_strides[0] = sizeof(in_data_type);
+    input_strides[1] = input_strides[0]* input_dims[0];
+    input_strides[2] = input_strides[1]* input_dims[1];
+    input_strides[3] = input_strides[2]* input_dims[2];
+    output_strides[0] = sizeof(out_data_type);
+    output_strides[1] = output_strides[0]* output_dims[0];
+    output_strides[2] = output_strides[1]* output_dims[1];
+    output_strides[3] = output_strides[2]* output_dims[2];
+#endif
+    Set4dTensorDesc(data->input_desc, in_data_type, input_dims,  input_strides);
+    Set4dTensorDesc(data->output_desc, out_data_type, output_dims,  output_strides);
+    if (parameters[2])
+      ERROR_CHECK_STATUS(vxCopyScalar((vx_scalar)parameters[2], &data->backend, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+
+    // Create custom lib and run setup
+    data->custom_handle = CustomCreate((CustomFunctionType)customFuntion);
+    ERROR_CHECK_CUSTOM_STATUS(CustomSetup(data->custom_handle, data->input_desc, data->output_desc, 
+                              (customBackend)data->backend, (customStream)data->hipstream));
+
+    ERROR_CHECK_STATUS(vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK uninitializeCustomLayer(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+    CustomLayerLocalData * data = NULL;
+    ERROR_CHECK_STATUS(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+    ERROR_CHECK_CUSTOM_STATUS(CustomShutdown(data->custom_handle));
```

Comment:
add a null check for data before accessing it

Reply:
done